webui/react/src/components/ProjectCard.tsx

```diff
@@ -1,104 +1,68 @@
 import { Typography } from 'antd';
-import React, { useCallback } from 'react';
+import React from 'react';
 
 import Tooltip from 'components/kit/Tooltip';
-import Link from 'components/Link';
 import TimeAgo from 'components/TimeAgo';
-import Avatar from 'components/UserAvatar';
 import { paths } from 'routes/utils';
 import Icon from 'shared/components/Icon/Icon';
-import { routeToReactUrl } from 'shared/utils/routes';
-import { useUsers } from 'stores/users';
-import { DetailedUser, Project } from 'types';
-import { Loadable } from 'utils/loadable';
+import { nearestCardinalNumber } from 'shared/utils/number';
+import { Project } from 'types';
 
-import ProjectActionDropdown from './ProjectActionDropdown';
+import Card from './kit/Card';
+import { useProjectActionMenu } from './ProjectActionDropdown';
 import css from './ProjectCard.module.scss';
 
 interface Props {
-  curUser?: DetailedUser;
   fetchProjects?: () => void;
   project: Project;
   showWorkspace?: boolean;
   workspaceArchived?: boolean;
 }
 
-const ProjectCard: React.FC<Props> = ({
-  project,
-  curUser,
-  fetchProjects,
-  workspaceArchived,
-  showWorkspace,
-}: Props) => {
-  const handleCardClick = useCallback(() => {
-    routeToReactUrl(paths.projectDetails(project.id));
-  }, [project.id]);
-
-  const users = Loadable.match(useUsers(), {
-    Loaded: (usersPagination) => usersPagination.users,
-    NotLoaded: () => [],
+const ProjectCard: React.FC<Props> = ({ project, fetchProjects, workspaceArchived }: Props) => {
+  const { menuProps, contextHolders } = useProjectActionMenu({
+    onComplete: fetchProjects,
+    project,
+    workspaceArchived,
   });
-  const user = users.find((user) => user.id === project.userId);
+
+  const classnames = [css.base];
+  if (project.archived) classnames.push(css.archived);
+
   return (
-    <ProjectActionDropdown
-      curUser={curUser}
-      project={project}
-      trigger={['contextMenu']}
-      workspaceArchived={workspaceArchived}
-      onComplete={fetchProjects}>
-      <div className={css.base} onClick={handleCardClick}>
-        <div className={css.nameRow}>
-          <h6 className={css.name}>
-            <Link inherit path={paths.projectDetails(project.id)}>
-              <Typography.Paragraph ellipsis={{ rows: 1, tooltip: true }}>
-                {showWorkspace && project.workspaceId !== 1 ? `${project.workspaceName} / ` : ''}
-                {project.name}
-              </Typography.Paragraph>
-            </Link>
-          </h6>
-          {project.archived && (
-            <Tooltip title="Archived">
-              <div>
-                <Icon name="archive" size="small" />
-              </div>
+    <Card
+      actionMenu={!project.immutable ? menuProps : undefined}
+      href={paths.projectDetails(project.id)}>
+      <div className={classnames.join(' ')}>
```

Comment:
Would it be possible to eliminate any of the custom CSS in these cards by using Grid from antd? It might help us start to understand what our layout support will need to look like in the UI Kit.

Reply:
I'll look into it.